### PR TITLE
[#IC-348] Add trigger to fill fiscalCode on message_status from message collection

### PR DESCRIPTION
**Pull request by fabriziopapi**
The message_status do not contains the fiscalCode. This will not allow us to query a message from its status (query with messageId only cost 8 time more). This trigger will fill all the existing message_status with the fiscalCode.

#### List of Changes

Add trigger on message collection to fill message_status fiscalCode

#### Motivation and Context

Allow to query a message from its message_status 

#### How Has This Been Tested?

system test

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by fabriziopapi** [comment]
In draft until io-function-commons merge

File: AddFiscalCodeToMessageStatus/index.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const contextTransport = new AzureContextTransport(() => logger, {
+  level: "debug"
+});
+winston.add(contextTransport);
+
+const messageStatusContainer = cosmosdbInstance.container(
+  MESSAGE_STATUS_COLLECTION_NAME
+);
+
+// const config = getConfigOrThrow();
```
Comment (michaeldisaro) [inline AddFiscalCodeToMessageStatus/index.ts:19]: Will this code be used? If not we can clean it.

**Review by michaeldisaro — APPROVED** [review]

File: AddFiscalCodeToMessageStatus/handler.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    rawMessages,
+    RA.map(
+      flow(
+        RetrievedMessageWithoutContent.decode,
+        TE.fromEither,
+        TE.chainW(patchAllVersion(container)),
+        TE.map(responses =>
+          pipe(
+            responses,
+            RA.reduce(404, (previous, result) =>
```
Comment (gquadrati) [inline AddFiscalCodeToMessageStatus/handler.ts:109]: Can you explain this reduce? If everything goes right, we get a 404? Is it fine?

**Review by gquadrati — CHANGES_REQUESTED** [review]

File: AddFiscalCodeToMessageStatus/handler.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+const getVersionedIds = (
+  messageId: string,
+  versionStart: number = 0,
+  maxBulk: number = 25 // Cosmos SDK limit = 100
+): ReadonlyArray<NonEmptyString> =>
+  RA.makeBy(
+    maxBulk,
+    i =>
+      `${messageId}-${(
+        "0".repeat(VERSION_PADDING_LENGTH) + String(versionStart + i)
```
[the remark this answers is not in the document]
Comment (fabriziopapi) [reply AddFiscalCodeToMessageStatus/handler.ts:33]: yes, but it works only with a model in input.

File: AddFiscalCodeToMessageStatus/index.ts (hunk 1 above)
Comment (fabriziopapi) [reply AddFiscalCodeToMessageStatus/index.ts]: fixed, thank you!

File: AddFiscalCodeToMessageStatus/handler.ts (hunk 2 above)
Comment (fabriziopapi) [reply AddFiscalCodeToMessageStatus/handler.ts:112]: fixed, thank you!

File: AddFiscalCodeToMessageStatus/handler.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            options
+          ),
+        toCosmosErrorResponse
+      ),
+    TE.filterOrElse(
+      results =>
+        pipe(
+          results,
+          AR.reduce(
+            true as boolean,
```
[the remark this answers is not in the document]
Comment (fabriziopapi) [reply AddFiscalCodeToMessageStatus/handler.ts:80]: fixed, thank you!

File: AddFiscalCodeToMessageStatus/handler.ts [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        operations: [
+          {
+            op: PatchOperationType.add,
+            path: `/fiscalCode`,
+            value: fiscalCode
+          }
+        ]
+      }
+    })),
+    readonlyPatchOperations => readonlyPatchOperations.slice(), // copy the readonly array to a mutable one
```
Comment (AleDore) [inline AddFiscalCodeToMessageStatus/handler.ts:64]:
```suggestion
    RA.toArray, // copy the readonly array to a mutable one
```

**Review by AleDore — APPROVED** [review]
LGTM

**Review by gquadrati — APPROVED** [review]

**Pull request merged**